stubs/docker/docker/utils/utils.pyi

```diff
@@ -1,34 +1,67 @@
-from _typeshed import Incomplete
-from typing import NamedTuple
+import datetime
+from _typeshed import FileDescriptorOrPath, Incomplete, ReadableBuffer
+from collections.abc import Iterable, Mapping
+from typing import Literal, NamedTuple, NoReturn, Protocol, TypeVar, overload, type_check_only
+
+from ..tls import TLSConfig
+
+_T = TypeVar("_T")
+_K = TypeVar("_K")
+_V = TypeVar("_V")
+
+# This Protocol is taken from stdlib/shlex.pyi
+@type_check_only
+class _ShlexInstream(Protocol):
+    def read(self, size: Literal[1], /) -> str: ...
+    def readline(self) -> object: ...
+    def close(self) -> object: ...
 
 class URLComponents(NamedTuple):
-    scheme: Incomplete
-    netloc: Incomplete
-    url: Incomplete
-    params: Incomplete
-    query: Incomplete
-    fragment: Incomplete
+    scheme: str | None
+    netloc: str | None
+    url: str
+    params: str | None
+    query: str | None
+    fragment: str | None
 
-def create_ipam_pool(*args, **kwargs) -> None: ...
-def create_ipam_config(*args, **kwargs) -> None: ...
-def decode_json_header(header): ...
-def compare_version(v1, v2): ...
-def version_lt(v1, v2): ...
-def version_gte(v1, v2): ...
-def convert_port_bindings(port_bindings): ...
-def convert_volume_binds(binds): ...
-def convert_tmpfs_mounts(tmpfs): ...
-def convert_service_networks(networks): ...
-def parse_repository_tag(repo_name): ...
-def parse_host(addr, is_win32: bool = False, tls: bool = False): ...
-def parse_devices(devices): ...
-def kwargs_from_env(environment: Incomplete | None = None): ...
-def convert_filters(filters): ...
-def datetime_to_timestamp(dt): ...
-def parse_bytes(s): ...
+def create_ipam_pool(*args, **kwargs) -> NoReturn: ...
+def create_ipam_config(*args, **kwargs) -> NoReturn: ...
```

Comment:
May as well include the deprecation message, since `NoReturn` is not enough to statically result in an error in all cases.
```suggestion
@deprecated("utils.create_ipam_pool has been removed. Please use a docker.types.IPAMPool object instead.")
def create_ipam_pool(*args, **kwargs) -> NoReturn: ...
@deprecated("utils.create_ipam_config has been removed. Please use a docker.types.IPAMConfig object instead.")
def create_ipam_config(*args, **kwargs) -> NoReturn: ...
```